src/ServiceControl.Monitoring.Data.Tests/RawDataReporterTests.cs

```diff
@@ -28,11 +30,59 @@ public void SetUp()
             buffer.TryWrite(3);
             buffer.TryWrite(4);
 
-            Assert(new long[]{1,2,3,4});
+            AssertValues(new long[]{1,2,3,4});
 
             await reporter.Stop();
         }
 
+        [Test]
+        public async Task When_flush_size_is_reached_only_maximum_number_is_flushed()
+        {
+            var reporter = new RawDataReporter(sender.ReportPayload, buffer, WriteEntriesValues, 4, TimeSpan.MaxValue);
+            reporter.Start();
+            buffer.TryWrite(1);
+            buffer.TryWrite(2);
+            buffer.TryWrite(3);
+            buffer.TryWrite(4);
+
+            AssertValues(new long[]{1,2}, new long[]{3,4});
+
+            await reporter.Stop();
+        }
+
+        [Test]
+        public async Task When_flushing_should_use_parallel_sends()
+        {
+            var payloads = new ConcurrentQueue<byte[]>();
```

Comment:
@danielmarbach Could you comment on this piece testing the parallelism of the reporter?